tools/roslaunch/src/roslaunch/nodeprocess.py

```diff
@@ -184,9 +193,20 @@ def __init__(self, run_id, package, name, args, env, log_output,
         @type  cwd: str
         @param is_node: (optional) if True, process is ROS node and accepts ROS node command-line arguments. Default: True
         @type  is_node: False
+        @param sigint_timeout: The SIGINT timeout used when killing nodes (in seconds).
+        @type sigint_timeout: int
+        @param sigterm_timeout: The SIGTERM timeout used when killing nodes if SIGINT does not stop the node (in seconds).
+        @type sigterm_timeout: int
         """    
         super(LocalProcess, self).__init__(package, name, args, env,
                 respawn, respawn_delay, required)
+
+        if sigint_timeout <= 0:
+            raise RLException("sigint_timeout must be a positive number, received %f" % sigint_timeout)
```

Comment:
Should this exception be documented in the docblock? Are there cases where that needs to bubble up the callstack and be documented there too?

Reply:
Done. I've propagated the docblock to all places I found this is recursively used.